classes/ActionScheduler_WPCLI_QueueRunner.php

```diff
@@ -0,0 +1,232 @@
+<?php
+
+/**
+ * WP CLI Queue runner.
+ *
+ * This class can only be called from within a WP CLI instance.
+ */
+class ActionScheduler_WPCLI_QueueRunner extends ActionScheduler_Abstract_QueueCleaner {
+
+	/** @var array */
+	protected $actions;
+
+	/** @var  ActionScheduler_ActionClaim */
+	protected $claim;
+
+	/** @var ActionScheduler_QueueCleaner */
+	protected $cleaner;
+
+	/** @var string */
+	protected $id;
+
+	/** @var ActionScheduler_FatalErrorMonitor */
+	protected $monitor;
+
+	/** @var \cli\progress\Bar */
+	protected $progress_bar;
+
+	/**
+	 * ActionScheduler_WPCLI_QueueRunner constructor.
+	 *
+	 * @param ActionScheduler_Store             $store
+	 * @param ActionScheduler_FatalErrorMonitor $monitor
+	 * @param ActionScheduler_QueueCleaner      $cleaner
+	 *
+	 * @throws Exception When this is not run within WP CLI
+	 */
+	public function __construct(
+		ActionScheduler_Store $store,
+		ActionScheduler_FatalErrorMonitor $monitor,
+		ActionScheduler_QueueCleaner $cleaner
+	) {
+		if ( ! ( defined( 'WP_CLI' ) && WP_CLI ) ) {
+			throw new Exception( __( 'The ' . __CLASS__ . ' class can only be run within WP CLI.', 'action-scheduler' ) );
+		}
+
+		$this->store   = $store;
+		$this->monitor = $monitor;
+		$this->cleaner = $cleaner;
```

Comment:
I like the use of dependency injection for the monitor and cleaner. Not essential, but if you're feeling generous, it's probably worth doing that consistently in the old `ActionScheduler_QueueRunner` now too. Though the 2nd and 3rd params on its `__construct()` will need to have default values set to be backward compatible.